kind/pkg/cluster/config/convert.go

```diff
@@ -0,0 +1,23 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+// ToCurrent converts Config to config.Config
+// It is implemented to meet config.Any, and just deep copies on this type
+func (c *Config) ToCurrent() *Config {
+	return c.DeepCopy()
```

Comment:
per out of band discussion: this is the only place `DeepCopy` is used currently. while it was fun to implement, we might want to just drop this functionality entirely, this method could just be `return c` for this type, and we shouldn't need `DeepCopy` elsewhere as we really shouldn't be mutating configs ...

Reply:
👍 for the time being - I think this depends on the direction taken with config handling/parsing etc. Things like converting between API versions and/or defaulting *may* require it in future.

Reply:
yeah, I'm not sure how heavily we need this. Happy to keep or remove or keep for now, or possibly switch to the upstream generators in a future PR if we think that's necessary.